drain/logging_memory.go

```
@@ -12,11 +12,16 @@ func NewLoggingMemory() *LoggingMemory {
 	}
 }
 
-func (l *LoggingMemory) Connect() bool {
-	return true
+func (l *LoggingMemory) Connect() error {
+	return nil
 }
 
-func (l *LoggingMemory) ShipEvents(fields map[string]interface{}, msg string) {
+func (l *LoggingMemory) Close() error {
+	return nil
+}
+
+func (l *LoggingMemory) ShipEvents(fields map[string]interface{}, msg string) error {
```

You added returning an error to this interface, but the implementations always return nil and the caller of ShipEvents ignores the error.